scripts/build.py

```diff
@@ -247,6 +247,15 @@ def download_sdk(conf, url, targetfolder, strip_components=1, force_extract=Fals
     else:
         print ("SDK already installed:", targetfolder)
 
+# a bit different function for xcode toolchain because package has slightly different structure
+def download_xcode_toolchain(conf, url, targetfolder, target_file_name, strip_components=1, force_extract=False, format='z'):
+    if not os.path.exists(join(targetfolder, target_file_name)) or force_extract:
+        path = conf.get_local_or_remote_file(url)
+        conf._extract_tgz_rename_folder(path, targetfolder, strip_components, format=format)
+    else:
+        print ("SDK already installed:", join(targetfolder, target_file_name))
```

Comment:
A bit unclear what's not functional with download_sdk.
Historically, we've maintained the packaging scripts so that the "download_sdk" function is easy to use: simply "unpack"

Reply:
Because in case of old approach we define a target folder which we use to unpack archive content. On case of Xcode now we need to unpack in the root of sdk folder (`ext/SDKs`). If I pass as target folder `ext/SDKs` it does nothing because it always exists. Also by passing target directory from outside of the function we can control how sdks are named ('rename').
So to address your comment I've just set `force_extract=True` for xcode toolchain and use the old `download_sdk` function.